### PR TITLE
feat(wallet): Portfolio Asset Segmented Controls

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Separates `Accounts` and `Transactions` with a `Segmented Controller` on the `Asset Details` view.
- Introduces `#` hash routing (`#accounts` or `#transactions`) will be applied to the end of `route` depending on which option you click on the `Segmented Controller`. If no `hash` is present, it will default to showing `Accounts`.

Resolves <https://github.com/brave/brave-browser/issues/30981>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Portfolio` page and select an asset to view
2. By default `Accounts` should be selected
3. Switch between `Accounts` and `Transactions` and make sure everything works properly
4. Observe the `#` value in the `Route`, make sure it updates depending on your selection.

https://github.com/brave/brave-core/assets/40611140/4ecb4f35-f542-4471-957f-189b04f133d5

### CODE REVIEW

File: components/brave_wallet_ui/components/shared/segmented-control/segmented-control.tsx [hunk 1]
```diff
@@ -20,20 +20,38 @@ import {
...
+      return useHashes
+        ? !hash
+          // Since hashes are not necessary
+          // we default the first option as selected,
+          // if there is no hash in the route location.
+          ? navOptions[0].id === navOption.id
+          : hash === navOption.hash
+        : walletLocation.includes(navOption.route)
+
+    }, [walletLocation, navOptions, useHashes, hash])
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/shared/segmented-control/segmented-control.tsx:54]: let's use relative navigation to handle this instead
ie: `history.push('#accounts')`

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx [hunk 2]
```diff
@@ -89,13 +97,19 @@ export const PortfolioAccountItem = (props: Props) => {
...
-  const orb = React.useMemo(() => {
-    return create({ seed: address.toLowerCase(), size: 8, scale: 16 }).toDataURL()
-  }, [address])
+  // Refs
+  const accountMenuRef = React.useRef<HTMLDivElement>(null)
 
+  // Hooks
+  useOnClickOutside(
+    accountMenuRef,
+    () => setShowAccountMenu(false),
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx]: Memoized!

File: components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx [hunk 3]
```diff
@@ -15,18 +14,28 @@ import {
...
 import Amount from '../../../utils/amount'
-import { getLocale } from '../../../../common/locale'
 import { getPriceIdForToken } from '../../../utils/api-utils'
 import { computeFiatAmount } from '../../../utils/pricing-utils'
 
 // Components
-import { TransactionPopup, WithHideBalancePlaceholder } from '../'
+import {
+  WithHideBalancePlaceholder
+} from '../'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx [hunk 4]
```diff
@@ -138,81 +220,166 @@ export const AccountsAndTransactionsList = ({
...
+                      address={account.address}
+                      accountKind={account.accountId.kind}
+                      assetBalance={getBalance(account, selectedAsset)}
+                      selectedNetwork={selectedAssetNetwork || selectedNetwork}
+                      showSellModal={() => onShowSellModal(account)}
+                      isSellSupported={checkIsAssetSellSupported(selectedAsset)}
+                      hideBalances={hidePortfolioBalances}
+                    />
+                  )
+                  }
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx]: Fixed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx [hunk 5]
```diff
@@ -138,81 +220,166 @@ export const AccountsAndTransactionsList = ({
...
+                    textColor='text03'
+                    isBold={false}
+                  >
+                    {
+                      getLocale('braveWalletNoAccountsWithABalanceDescription')
+                    }
+                  </Text>
+                </Column>
+              )
+              }
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx]: Fixed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/style.ts
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/style.ts:1]: Was fixed earlier, must have been looking at the first commit change! :)

File: components/brave_wallet_ui/components/desktop/portfolio-account-item/style.ts [hunk 6]
```diff
@@ -67,56 +63,50 @@ export const BalanceColumn = styled.div`
...
+  pointer-events: auto;
   border: none;
 `
 
-export const MoreIcon = styled(MoreVertRIcon)`
-  width: auto;
-  height: 26px;
-  transform: rotate(90deg);
-  color: ${(p) => p.theme.color.interactive08};
+export const AccountMenuIcon = styled(Icon)`
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/portfolio-account-item/style.ts]: Updated!

File: components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx [hunk 7]
```diff
@@ -156,32 +169,37 @@ export const PortfolioAccountItem = (props: Props) => {
...
-              text={getLocale('braveWalletTransactionExplorer')}
+        <AccountMenuWrapper
+          ref={accountMenuRef}
+        >
+          <AccountMenuButton
+            onClick={() => setShowAccountMenu(prev => !prev)}
+          >
+            <AccountMenuIcon
+              name='more-vertical'
             />
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/portfolio-account-item/index.tsx]: Updated!

File: components/brave_wallet_ui/components/shared/segmented-control/segmented-control.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/shared/segmented-control/segmented-control.tsx]: Updated this with you! :)

**Review by josheleonard — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings` LGTM

**Review by darkdh — APPROVED** [review]

**Pull request merged**